src/isl/isl_type_reference.rs

```diff
@@ -103,45 +84,98 @@ impl IslTypeRef {
+            "$float" => type_store.get_builtin_type_id(&format!("${}", Float)),
+            "$decimal" => type_store.get_builtin_type_id(&format!("${}", Decimal)),
+            "$timestamp" => type_store.get_builtin_type_id(&format!("${}", Timestamp)),
+            "$string" => type_store.get_builtin_type_id(&format!("${}", String)),
+            "$symbol" => type_store.get_builtin_type_id(&format!("${}", Symbol)),
+            "$bool" => type_store.get_builtin_type_id(&format!("${}", Boolean)),
+            "$blob" => type_store.get_builtin_type_id(&format!("${}", Blob)),
+            "$clob" => type_store.get_builtin_type_id(&format!("${}", Clob)),
+            "$sexp" => type_store.get_builtin_type_id(&format!("${}", SExpression)),
+            "$list" => type_store.get_builtin_type_id(&format!("${}", List)),
+            "$struct" => type_store.get_builtin_type_id(&format!("${}", Struct)),
+            "$any" => type_store.get_builtin_type_id("$any"),
+            "$text" => type_store.get_builtin_type_id("$text"),
+            "$lob" => type_store.get_builtin_type_id("$lob"),
+            "$number" => type_store.get_builtin_type_id("$number"),
+            // TODO: add a match for document builtin type
+            _ => {
                 // verify if the AliasType actually exists in the type_store or throw an error
                 match pending_types.get_type_id_by_name(alias, type_store) {
-                    Some(type_id) => Ok(type_id.to_owned()),
+                    Some(type_id) => Some(type_id.to_owned()),
                     None => match pending_types.get_parent() {
                         Some(parent) => {
                             // if it is a self referencing type resolve it using parent information from type_store
                             if parent.0.eq(alias) {
-                                Ok(parent.1)
+                                Some(parent.1)
                             } else {
-                                unresolvable_schema_error(format!(
+                                return unresolvable_schema_error(format!(
                                     "Could not resolve type reference: {:?} does not exist",
                                     alias
-                                ))
+                                ));
                             }
                         }
-                        None => unresolvable_schema_error(format!(
-                            "Could not resolve type reference: {:?} does not exist",
-                            alias
-                        )),
+                        None => {
+                            return unresolvable_schema_error(format!(
+                                "Could not resolve type reference: {:?} does not exist",
+                                alias
+                            ))
+                        }
                     },
                 }
             }
+        }
+        .ok_or(invalid_type_reference_error)
```

Comment:
The match condition for all of the built-in types is a huge wall of essentially repeated code, and it means that we have more than one place where we need to make sure that all of the built-in types are enumerated correctly. Since the built-in types are preloaded, we know that they will already be in the `type_store`, so rather than matching on type names, would something like this (rust-ish) code sample work?

```rust
return if let Some(type_id) = type_store.get_built_in_type(alias) {
    type_id
} else if let Some(type_id) = pending_types.get_type_id_by_name(alias, type_store) {
    type_id
} else if let Some(parent) = pending_types.get_parent() && parent.0.eq(alias) {
    parent.1
} else {
    invalid_type_reference_error = unresolvable_schema_error_raw(format!(
        "Could not resolve type reference: {:?} does not exist",
        alias
    ))
}
```

If we absolutely must match the name before we try to get it from the `type_store`, then we should so something like this instead of having a match condition for each possible type name:
```rust
const BUILT_IN_TYPE_NAMES: &[&str] = &["$string", "string" /* etc. */];

// ...

return if BUILT_IN_TYPE_NAMES.into_iter().contains(alias) {
    type_store.get_built_in_type(alias)
} //...
```